beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/filter/IcebergTableListenerEventFilter.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (C) 2019-2024 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.scheduler.apiary.filter;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+import com.expedia.apiary.extensions.receiver.common.event.ListenerEvent;
+import com.expediagroup.beekeeper.core.model.LifecycleEventType;
+
+import java.util.Locale;
+import java.util.Map;
+
+@Component
+public class IcebergTableListenerEventFilter implements ListenerEventFilter {
+
+  private static final Logger log = LogManager.getLogger(IcebergTableListenerEventFilter.class);
+
+  private static final String METADATA_LOCATION_KEY = "metadata_location";
+  private static final String TABLE_TYPE_KEY = "table_type";
+  private static final String TABLE_TYPE_ICEBERG_VALUE = "iceberg";
+
+  @Override
+  public boolean isFiltered(ListenerEvent event, LifecycleEventType type) {
+    Map<String, String> tableParameters = event.getTableParameters();
+
+    if (tableParameters != null) {
```

Comment:
refactor this logic in a separate class please so we have the checks in one place.

The same check is here: https://github.com/ExpediaGroup/beekeeper/pull/185/files#diff-4193e6f6cfd4a695a7f5f655f79dece56e839ec14f5e19e839102d55665970bfR47

You can make an `IsIcebergPredicate implements Predicate<Map<String, String>>` -> [java/util/function/Predicate](https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html)